### PR TITLE
feat: refactor data explorer and add XDD facets support

### LINKED ISSUE #178: [TASK]: Clean up the `Data Explorer` to integrate it in the HMI
**Describe the task**
* As of now the `Data Explorer` has been implemented as a research phase.
* Remove all notions of World Modellers from it
* Re-design UI to look and feels like TERArium
* Change Vue3 to Vue Script Setup language

**Goals**
* Make the Data Explorer seamlessly integrate within the HMI

### PR DESCRIPTION
**Pull request by amostafa-uncharted**
# Description

- Refactor components utilized by the dat explorer to match the Vue Script Setup component style.
- Add frontend and backend support for using facets provided by the XDD service
- Other misc bug fixes and style improvements
- Integrate the usage of axios and remove the uncloak function 
- add support for fetching related documents from xdd

**NOTE**:
This PR attempt to refactor almost all components used by the data explorer and hence too many files are touched. However, please note that most of the refactoring is mainly to rewrite each component in the Vue style with the script setup tag. While I do not honestly find such a refactoring needed, it has been a long standing requirement from @YohannParis and the associated ticket in favor of consistency. 
For easier review, you may focus on the changes made prior to the refactoring effort, i.e., ignore the last 3 commits or so and examine the changes up to that point.
 
Resolves #178 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide reproduction instructions and list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by chris-dickson — COMMENTED** [review]
Seems reasonable.  What's the plan for exceptions though?  Is the service layer expected to handle API exceptions or is it one more level up?  @dgauldie ?

**Comment by chris-dickson** [comment]
@amostafa-uncharted can you provide a bit more detail on the changes here?  36 files is non trivial and as a reviewer I don't know where to start.

**Comment by amostafa-uncharted** [comment]
> @amostafa-uncharted can you provide a bit more detail on the changes here? 36 files is non trivial and as a reviewer I don't know where to start.

This PR attempt to refactor almost all components used by the data explorer and hence too many files are touched. However, please note that most of the refactoring is mainly to rewrite each component in the Vue style with the script setup tag. While I do not honestly find such a refactoring needed, it has been a long standing requirement from @YohannParis in favor of consistency. You may focus on the changes made prior to the refactoring effort, i.e., ignore the last 3 commits or so and examine the changes up to that point. I'll try to add more details at the the description of the PR.

File: packages/client/hmi-client/src/types/Filter.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// WM search construct
+// earch construct
```
Comment (YohannParis) [inline packages/client/hmi-client/src/types/Filter.ts:1]:
```suggestion
// search construct
```

File: packages/client/hmi-client/src/components/widgets/autocomplete.vue [hunk 2]
```diff
@@ -1,3 +1,5 @@
+import { computed } from "vue"; import { computed } from "vue";
+
```
Comment (YohannParis) [inline packages/client/hmi-client/src/components/widgets/autocomplete.vue:2]: This should not be there, maybe a pasting mistake.

File: packages/client/hmi-client/src/components/widgets/dropdown-button.vue [hunk 3]
```diff
@@ -43,130 +43,117 @@ import IconCheckmark16 from '@carbon/icons-vue/es/checkmark/24';
 export type DropdownItem = {
 	displayName: string;
 	// eslint-disable-next-line @typescript-eslint/no-explicit-any
-	value: any;
+	value: string;
```
Comment (YohannParis) [inline packages/client/hmi-client/src/components/widgets/dropdown-button.vue:46]: The eslint-disable comment can be removed.

File: packages/client/hmi-client/src/components/data-explorer/simple-pagination.vue [hunk 4]
```diff
@@ -1,3 +1,5 @@
+import { computed } from "vue";
+
```
Comment (YohannParis) [inline packages/client/hmi-client/src/components/data-explorer/simple-pagination.vue:2]: Should not be there

**Review by YohannParis — COMMENTED** [review]
Thank you for moving all those components to _script setup_.
Also I really appreciate all the useful comments that you add in your code! 🙏

**Review by chris-dickson — APPROVED** [review]

File: packages/client/hmi-client/src/components/widgets/dropdown-button.vue (hunk 3 above)
Comment (amostafa-uncharted) [reply packages/client/hmi-client/src/components/widgets/dropdown-button.vue:46]: done

File: packages/client/hmi-client/src/components/widgets/autocomplete.vue (hunk 2 above)
Comment (amostafa-uncharted) [reply packages/client/hmi-client/src/components/widgets/autocomplete.vue:2]: done

**Review by YohannParis — APPROVED** [review]

**Review by chris-dickson — APPROVED** [review]

**Review by YohannParis — APPROVED** [review]

**Review by dgauldie — APPROVED** [review]
LGTM

**Pull request merged**